props/saber_fett263_buttons.h

```diff
@@ -293,248 +660,3355 @@ class SaberFett263Buttons : public virtual PropBase {
+            menu_type_ = MENU_RESET_COLOR;
+            PlayMenuSound("mconfirm.wav");
+            break;
+          default:
+            PlayMenuSound("moption.wav");
+        }
+  #elif NUM_BLADES == 2
+        switch (menu_sub_pos_) {
+          case 1:
+            menu_type_ = MENU_EFFECT;
+            PlayMenuSound("meffect.wav");
+            break;
+          case 2:
+            menu_type_ = MENU_COPY_COLOR;
+            SetColorCopy();
+            PlayMenuSound("mconfirm.wav");
+            break;
+          case 3:
+            menu_type_ = MENU_RESET_COLOR;
+            PlayMenuSound("mconfirm.wav");
+            break;
+          default:
+            PlayMenuSound("moption.wav");
+            break;
+        }
+  #else 
+        switch (menu_sub_pos_) {
+          case 1:
+            menu_type_ = MENU_EFFECT;
+            PlayMenuSound("meffect.wav");
+            break;
+          case 2:
+            menu_type_ = MENU_BLADE_COPY;
+            SetColorCopy();
+            PlayMenuSound("mblade.wav");
+            break;
+          case 3:
+            menu_type_ = MENU_RESET_COLOR;
+            PlayMenuSound("mconfirm.wav");
+            break;
+          default:
+            PlayMenuSound("moption.wav");
+        }
+  #endif
+        break;
+      case MENU_COLOR_MODE:
+        edit_color_ = true;
+        switch (sub_dial_) {
+          case 0:
+            twist_menu_ = M_PI / 6;
+            dial_ = -1;
+            color_mode_ = COLOR_LIST;
+            break;
+          case 1:
+            color_mode_ = EDIT_COLOR;
+            hsl_.S = 1.0;
+            break;
+          case 2: 
+            color_mode_ = EDIT_WHITE;
+            twist_menu_ = M_PI / 18;
+            hsl_.S = 1.0;
+            break;
+          case 3: 
+            color_mode_ = EDIT_BLACK;
+            twist_menu_ = M_PI / 36;
+            hsl_.S = 1.0;
+            break;
+          default:
+            break;
+        }
+        switch (effect_num_) {
+          case BASE_COLOR_ARG:
+            menu_type_ = COLOR_BASE;
+            ShowFull();
+            break;
+          case ALT_COLOR_ARG:
+            menu_type_ = COLOR_ALT;
+            ShowFull();
+            break;
+          case BLAST_COLOR_ARG:
+            menu_type_ = COLOR_BLAST;
+            bump_color_.Start(blade_num_, SkipWord(SkipWord(current_preset_.GetStyle(blade_num_))));                 
+            ShowPart();
+            break;
+          case CLASH_COLOR_ARG:
+            menu_type_ = COLOR_CLASH;
+            bump_color_.Start(blade_num_, SkipWord(SkipWord(current_preset_.GetStyle(blade_num_))));                 
+            ShowPart();
+            break;
+          case LOCKUP_COLOR_ARG:
+            menu_type_ = COLOR_LOCKUP;
+            bump_color_.Start(blade_num_, SkipWord(SkipWord(current_preset_.GetStyle(blade_num_))));                 
+            ShowPart();
+            break;
+          case DRAG_COLOR_ARG:
+            menu_type_ = COLOR_DRAG;
```

Comment:
Do we really need to have a menu type for each type of argument we can edit?
Seems like it would be easier to just have one (or a few) menu types and then use the effect_num_ to if we need to do anything different in a few places?

Reply:
The menu controls change for each menu_type_, the effect_num_ is the argument value but selection, setting and values are based on the menu you're in.

Reply:
but does it have to work like that?

Reply:
Noted in other comment, seems like it'd require a pretty significant rewrite to change the menu implementation and I'm not sure it's within my abilities.  I attempted several iterations to set up sub-menu navigation as it's own thing but none worked properly or actually made the switch statements even more complicated for the dial and button functionality.  This approach has been working very well for the entire time I've been testing and developing.